rotkehlchen/exchanges/bybit.py

```diff
@@ -0,0 +1,535 @@
+                    else:
+                        usd_price = Inquirer().find_usd_price(asset=asset)
+                        usd_value = usd_price * amount
+                except (UnknownAsset) as e:
+                    self.msg_aggregator.add_warning(
+                        f'Found Bybit balance result with unknown asset '
+                        f'{e.identifier}. Ignoring it.',
+                    )
+                    continue
+                except (DeserializationError, KeyError) as e:
+                    msg = str(e)
+                    if isinstance(e, KeyError):
+                        msg = f'Missing key entry for {msg}.'
+                    return None, f'Error processing Bybit balance entry {coin_data}. {msg}'
+
+                assets_balance[asset] += Balance(amount=amount, usd_value=usd_value)
+
+        return assets_balance, ''
+
+    def _query_deposits_withdrawals(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+            query_for: AssetMovementCategory,
+    ) -> list[AssetMovement]:
+        """
+        Process deposits from bybit. If any asset is unknown or we fail to deserialize an entry
+        the error is logged and we skip the entry.
+
+        This logic doesn't filter by timestamp the query and instead process all the entries
+        returned by the api discarding those that aren't in start_ts <= timestamp <= end_ts.
+        The reason is that the api is poorly designed and you can't know until when to query,
+        for any time range the api will return values and is not possible to know when the user
+        started to use the api.
+        """
+        log.debug(f'querying bybit online {query_for} with {start_ts=}-{end_ts=}')
+        if query_for == AssetMovementCategory.DEPOSIT:
+            endpoint = 'asset/deposit/query-record'
+            timestamp_key = 'successAt'
+        else:
+            endpoint = 'asset/withdraw/query-record'
+            timestamp_key = 'updateTime'
+
+        raw_data = self._paginated_api_query(
+            endpoint=endpoint,  # type: ignore  # mypy doesn't detect that the string is assigned once
+            options={'limit': PAGINATION_LIMIT},
+        )
+
+        movements = []
+        for movement in raw_data:
+            if (timestamp_raw := movement.get(timestamp_key)) is not None:
+                timestamp = ts_ms_to_sec(TimestampMS(int(timestamp_raw)))
+                if not start_ts <= timestamp <= end_ts:
+                    continue  # skip if is not in range
+            else:
+                continue  # skip missing timestamp
+
+            try:
+                coin = asset_from_bybit(movement['coin'])
+            except UnknownAsset:
+                log.error(f'Could not read assets from trade pair {movement["coin"]}')
+                continue
+
+            try:
+                if query_for == AssetMovementCategory.DEPOSIT:
```

Comment:
instead of an extra if check for every single loop iteration make these dict keys parametrizable and assign them at the start of the function where you already do this if check